internal/clients/vault_secrets.go

```diff
@@ -118,19 +122,32 @@ func CreateVaultSecretsAppSecret(ctx context.Context, client *Client, loc *share
 
 // OpenVaultSecretsAppSecret will retrieve the latest secret for a Vault Secrets app, including it's value.
 func OpenVaultSecretsAppSecret(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, appName, secretName string) (*secretmodels.Secrets20230613OpenSecret, error) {
-
 	getParams := secret_service.NewOpenAppSecretParams()
 	getParams.Context = ctx
 	getParams.AppName = appName
 	getParams.SecretName = secretName
 	getParams.LocationOrganizationID = loc.OrganizationID
 	getParams.LocationProjectID = loc.ProjectID
 
-	getResp, err := client.VaultSecrets.OpenAppSecret(getParams, nil)
-	if err != nil {
-		return nil, err
+	var getResp *secret_service.OpenAppSecretOK
+	var err error
+	for attempt := 0; attempt < retryCount; attempt++ {
+		getResp, err = client.VaultSecrets.OpenAppSecret(getParams, nil)
+		if err != nil {
+			serviceErr, ok := err.(*secret_service.OpenAppSecretDefault)
+			if !ok {
+				return nil, err
+			}
+			if shouldRetryErrorCode(serviceErr.Code(), []int{429}) {
+				backOffDuration := getAPIBackoffDuration(serviceErr)
+				fmt.Printf("The api rate limit exceeded, attempt: %d trying in %d seconds", (attempt + 1), int64(backOffDuration.Seconds()))
```

Comment:
Maybe I'm missing it, but i'm not seeing this warning message in the plan output in the description 🤔 

also minor nit on wording, but feel free to ignore that
`The api rate limit has been exceeded, trying in %d seconds, attempt: %d`

Reply:
thanks, will incorporate the feedback. for the log line question, need to set export TF_LOG=DEBUG. I'll add a trace with that as well.

Reply:
Added trace output to the PR description.